### ISSUE TITLE
Build is working with Arduino but fails with PlatformIO

### ISSUE THREAD

**Issue by chaudhryfaisal** [ubis/STM32duino-PlatformIO#7]
I have issues when I build blinky code with PlatformIO. Same code works fine when I build with Arduino and flash using maple_upload util

`#include <Arduino.h>
#define LED1 PC13
bool on = false;
void setup() {
  pinMode(LED1, OUTPUT);
}

void loop() {
  digitalWrite(LED1, on?HIGH:LOW);
  on = !on;
  delay(1000);       
}
`

Log when flashing with arduino

`"C:\Users\USER\AppData\Local\Arduino15\packages\arduino\tools\arm-none-eabi-gcc\4.8.3-2014q1/bin/arm-none-eabi-objcopy" -O binary  "C:\Users\USER\AppData\Local\Temp\build9c9ef3bdfe2fccb480bc6e4bac749e41.tmp/Blink.ino.elf" "C:\Users\USER\AppData\Local\Temp\build9c9ef3bdfe2fccb480bc6e4bac749e41.tmp/Blink.ino.bin"

Sketch uses 12,892 bytes (19%) of program storage space. Maximum is 65,536 bytes.
Global variables use 2,800 bytes of dynamic memory.
C:\Users\USER\Documents\Arduino\hardware\Arduino_STM32-master/tools/win/maple_upload.bat COM24 2 1EAF:0003 C:\Users\USER\AppData\Local\Temp\build9c9ef3bdfe2fccb480bc6e4bac749e41.tmp/Blink.ino.bin 
maple_loader v0.1
Resetting to bootloader via DTR pulse
Reset via USB Serial Failed! Did you select the right serial port?
Assuming the board is in perpetual bootloader mode and continuing to attempt dfu programming...
Searching for DFU device [1EAF:0003]...

Found it!

Opening USB Device 0x1eaf:0x0003...
Found Runtime: [0x1eaf:0x0003] devnum=1, cfg=0, intf=0, alt=2, name="STM32duino bootloader v1.0  Upload to Flash 0x8002000"
Setting Configuration 1...
Claiming USB DFU Interface...
Setting Alternate Setting ...
Determining device status: state = dfuIDLE, status = 0
dfuIDLE, continuing
Transfer Size = 0x0400
bytes_per_hash=257
Starting download: [##################################################] finished!
error resetting after download: usb_reset: could not reset device, win error: The system cannot find the file specified.
state(8) = dfuMANIFEST-WAIT-RESET, status(0) = No error condition is present

Done!

Resetting USB to switch back to runtime mode
`

Log when flashing with PlatformIO

`Linking .pioenvs\f103c8\firmware.elf
Checking program size .pioenvs\f103c8\firmware.elf
text       data     bss     dec     hex filename
12904      1928     896   15728    3d70 .pioenvs\f103c8\firmware.elf
arm-none-eabi-objcopy -O binary .pioenvs\f103c8\firmware.elf .pioenvs\f103c8\firmware.bin
BeforeUpload(["upload"], [".pioenvs\f103c8\firmware.bin"])
Use manually specified: COM4
C:\Users\Owner\.platformio\packages\framework-stm32duino\tools\win\maple_upload COM4 2 1EAF:0003 D:\Development\_Svn\gitlab\mi
crocontroller\arm\PlatformIO\STM32F103C8T6\A_Blink/.pioenvs\f103c8\firmware.bin
maple_loader v0.1
Resetting to bootloader via DTR pulse

Searching for DFU device [1EAF:0003]...
Assuming the board is in perpetual bootloader mode and continuing to attempt dfu programming...
 
Found it!
 
Opening USB Device 0x1eaf:0x0003...

Found Runtime: [0x1eaf:0x0003] devnum=1, cfg=0, intf=0, alt=2, name="STM32duino bootloader v1.0  Upload to Flash 0x8002000"
Setting Configuration 1...
Claiming USB DFU Interface...
Setting Alternate Setting ...
Determining device status: state = dfuIDLE, status = 0
dfuIDLE, continuing
Transfer Size = 0x0400
bytes_per_hash=296
Starting download: [##################################################] finished!
error resetting after download: usb_reset: could not reset device, win error: The system cannot find the file specified.
 
 
state(8) = dfuMANIFEST-WAIT-RESET, status(0) = No error condition is present
Done!
Resetting USB to switch back to runtime mode
 [SUCCESS] Took 7.60 seconds`

I did notice compiler is different between two IDEs and size produced is also different. 
Arduino is using "arm-none-eabi-gcc\4.8.3-2014q1" and PlatformIO is using "arm-none-eabi/4.8.4/"

**Comment by ubis** [comment]
I'll start write changes and creating PR in [platform-ststm32](https://github.com/platformio/platform-ststm32), hopefully I will fix all of errors.

And yeah, PlatformIO uses newer gcc compiler, but that shouldn't be the case.

**Comment by chaudhryfaisal** [comment]
I am sorry, I didn't quiet get your comment. You want me to create a Pull Request in the platform-ststm32? What errors will you fix? I dont get any errors my whole build + upload is success just my led doesnt blink.

**Comment by ubis** [comment]
No, this repo(stm32duino-platformio) will be merged into official PlatformIO ststm32 platform by me.
There are some things 'underhood' that must be fixed/changed/replaced, probably some of that are causing your build to not work properly.

Sorry, I actually didn't read the whole comment, since I won't be doing any changes to this repo anymore, I just posted my comment from another issue.

**Comment by chaudhryfaisal** [comment]
got it, I was really hoping I would get this working but I guess I will just have to wait. 

Thank you so much for all of your hardwork

**Comment by ubis** [comment]
STM32duino is now officialy supported by PlatformIO 3.0: [https://github.com/platformio/platform-ststm32](https://github.com/platformio/platform-ststm32), install and check if your code now works.
If not, post a issue in that repo, this one will be closed soon.
